provider/kubernetes/kubernetes.go

```diff
@@ -190,6 +190,18 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
 		}
 		templateObjects.TLS = append(templateObjects.TLS, tlsSection...)
 
+		var weightAllocator weightAllocator = &defaultWeightAllocator{}
+		annotationPercentageWeights := getAnnotationName(i.Annotations, annotationKubernetesServiceWeights)
+		if _, ok := i.Annotations[annotationPercentageWeights]; ok {
+			fractionalAllocator, err := newFractionalWeightAllocator(i, k8sClient)
+			if err != nil {
+				log.Warnf("Failed to read backend percentage weights for ingress %s/%s: %v", i.Namespace, i.Name, err)
+				weightAllocator = &defaultWeightAllocator{}
```

Comment:
I guess the argument is that a user editing the weights could completely disable the ingress with an error in the annotation, whereas the fallback would allow the ingress to continue with the default weights. Preference IMO.

Reply:
I would be ok with falling back to default weights if the annotation is invalid

Reply:
The dreadful scenario I'm thinking of is that I intend to serve 1% of traffic to my canary release, but due to a syntax bug in the annotation Traefik suddenly serves 100%, thereby overloading the application. A 404 served from Traefik because the frontend isn't even registered seems easier to detect/debug.

Reply:
I can agree with that logic